src/main/java/graphql/annotations/processor/ProcessingElementsContainer.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.processor;
+
+
+import graphql.annotations.processor.typeFunctions.TypeFunction;
+import graphql.relay.Relay;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+public class ProcessingElementsContainer {
+
+    private TypeFunction defaultTypeFunction;
+    private graphql.relay.Relay relay;
+    private Map<String, graphql.schema.GraphQLType> typeRegistry;
+    private Map<Class<?>, Set<Class<?>>> extensionsTypeRegistry;
+    private Stack<String> processing;
+
+
+    public ProcessingElementsContainer(TypeFunction defaultTypeFunction, Relay relay, Map<String, graphql.schema.GraphQLType> typeRegistry, Map<Class<?>, Set<Class<?>>> extensionsTypeRegistry, Stack<String> processing) {
```

Comment:
I think it would be more clear to skip the 3 last parameters and directly initialize with empty map/stacks in the constructor